### PR TITLE
containsExactly on all values for a session key

### PR DESCRIPTION
**Pull request by yanns**

### CODE REVIEW

File: cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala [hunk 1]
```diff
@@ -129,6 +134,27 @@ object SessionSteps {
     def asJson: JsonStepBuilder = JsonStepBuilder(sessionKey, prettySessionKeyTitle)
   }
 
+  case class SessionArrayStepBuilder(private val sessionKey: String) {
+    def containsExactly[A: Resolvable](elements: A*): AssertStep = {
+      val prettyElements = elements.mkString(" and ")
+      val title = s"$sessionKey value contains\n$prettyElements"
```
Comment (agourlay) [inline cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala:140]: `s"$sessionKey history contains exactly\n$prettyElements"`

**Review by agourlay — COMMENTED** [review]
Thank you for your contributions!

I have only one concern about the naming of the DSL function, the rest looks :ok_hand:

File: cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/CoreDsl.scala [hunk 2]
```diff
@@ -166,6 +166,9 @@ trait CoreDsl {
   def session_values(k1: String, k2: String): SessionValuesStepBuilder =
     SessionValuesStepBuilder(k1, k2)
 
+  def session_values(key: String): SessionArrayStepBuilder =
```
[the remark this answers is not in the document]
Comment (yanns) [reply cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/CoreDsl.scala]: https://github.com/agourlay/cornichon/pull/495/commits/b8b1c6e3c918ace0d1753c1421df66579083d97c

File: cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala [hunk 3]
```diff
@@ -129,6 +134,27 @@ object SessionSteps {
...
+      bodyContainsExactlyElmt(title, elements)
+    }
+
+    private def bodyContainsExactlyElmt[A: Resolvable](title: String, expectedElements: Seq[A]) =
```
[the remark this answers is not in the document]
Comment (yanns) [reply cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala]: https://github.com/agourlay/cornichon/pull/495/commits/b8b1c6e3c918ace0d1753c1421df66579083d97c

File: cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala (hunk 1 above)
Comment (yanns) [reply cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/SessionSteps.scala]: https://github.com/agourlay/cornichon/pull/495/commits/b8b1c6e3c918ace0d1753c1421df66579083d97c

**Pull request merged**